### ISSUE TITLE
Library click through to template part focus mode hides sidebar and Navigation editing feature

### ISSUE THREAD

**Issue by getdave** [WordPress/gutenberg#51839]
As part of Navigation on Browse Mode it was agreed that in focus mode for Template Parts it would be useful to [display any Navigation Menus present in the blocks within in the browse mode sidebar](https://github.com/WordPress/gutenberg/pull/51492).

However with the change to Library, when you click on a Template Part you now get taken to the focus mode in "edit" mode (`canvas=edit`) which hides the sidebar.

Given the value of https://github.com/WordPress/gutenberg/pull/51492, we should retain the sidebar. If necessary this can be done conditionally by detecting whether a Navigation block is present in the blocks.

https://github.com/WordPress/gutenberg/assets/444434/3f0d2666-697b-40da-98f1-7c22564f3aaf

**Comment by getdave** [comment]
cc @talldan for visibility

**Comment by talldan** [comment]
> cc @talldan for visibility

Thanks. I don't know much about the sidebar specifics. It might have been a design choice or an accidental inclusion where we didn't spot the regression. It's quite a subtle difference, so when testing I personally wouldn't have considered this a bug.

@SaxonF @kevin940726 or @aaronrobertshaw may know more.

**Comment by getdave** [comment]
@WordPress/gutenberg-design Just a reminder that this feature was requested in https://github.com/WordPress/gutenberg/issues/50704 as part of https://github.com/WordPress/gutenberg/issues/50396.

I think all that needs to change is, if there is a Navigation in the Template Part then show the sidebar. 

Or we could just show the sidebar at all times.

**Comment by kevin940726** [comment]
I noticed this too when I was fixing the routing issue for mobile. I thought it was a deliberate change so I retained the behavior. I agree that landing the `canvas=view` page first will be better though!

FWIW, due to how the sidebar animation is working now, landing directly to the edit mode might cause some unwanted flickering. It's not impossible to fix, but landing to the view mode first might make it easier.
